### PR TITLE
🐛 Source Hubspot: Removed stage history from deals stream

### LINKED ISSUE #10544: HubSpot Sync Job Hangs on Deals Sync
## Environment
- **Airbyte version**: 0.35.32-alpha
- **OS Version / Instance**: Ubuntu Focal 20.04
- **Deployment**: EC2
- **Source Connector and version**: HubSpot 0.1.41	
- **Destination Connector and version**: Redshift 0.3.25	
- **Severity**: High
- **Step where error happened**: Sync job

## Current Behavior
Airbyte sync hangs on `Syncing stream: deals`, syncs succeed without deals

## Expected Behavior
Airbyte should sync deals from HubSpot

## Logs
*If applicable, please upload the logs from the failing operation. 
For sync jobs, you can download the full logs from the UI by going to the sync attempt page and 
clicking the download logs button at the top right of the logs display window.*

<details>
<summary>LOG</summary> 

2022-02-22 14:36:36 source > Finished syncing SourceHubspot
2022-02-22 14:36:36 source > SourceHubspot runtimes:
2022-02-22 14:36:36 source > Syncing stream: deal_pipelines 
2022-02-22 14:36:36 source > Read 3 records from deal_pipelines stream
2022-02-22 14:36:36 source > Finished syncing SourceHubspot
2022-02-22 14:36:36 source > SourceHubspot runtimes:
2022-02-22 14:36:36 source > Syncing stream: deals 

</details>

## Steps to Reproduce
1. Setup HubSpot source with API key and deals as an input
2. Setup Redshift destination with S3 copying
3. Run a sync job

## Are you willing to submit a PR?
Yes

### PR DESCRIPTION
**Pull request by lgomezm**
## What

- Fixes: https://github.com/airbytehq/airbyte/issues/10544
- The deal stream currently gets all deal stage history: https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py#L1046-L1048
- This in turn gets all deals: https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py#L999-L1006
- That is why deal jobs hang. For full syncs, it will get all deals twice.
- Even on getting incremental updates, it will get all deals (to get the stage history), to then start yielding only the new/updated deals.

## How

- I have removed, from the `Deals` stream, the code that gets all deal stage history records.
- I believe getting all records from another stream to set one property is not a good practice.
- Some other relationships work if one of the records includes the ID of the associated object (i.e. for the relationship between Hubspot contacts and companies, they have the associated object ID).
- Something like that can be done, at the destination, with deals and deal stage history.

## Recommended reading order
1. `streams.py`
2. `test_source.py`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Comment by lgomezm** [comment]
> @lgomezm it's possible to show the integration tests results? After doing this you're able to run the deals stream?

@marcosmarxm Here they are. I had to update the `CRMSearchStream` class. They were iterating on the super class' slices, which is not necessary for that kind of stream -- It was taking too long if you used a very old `start_date` in your config. See [1a0596c](https://github.com/airbytehq/airbyte/pull/10707/commits/1a0596ca650bd8789bb8ea6932558e5989e84dea) for reference.
Screen Shot 2022-03-03 at 8 58 14 PM

**Comment by lgomezm** [comment]
@marcosmarxm I'm also able to run the deals stream successfully. The acceptance tests already do that both as full refresh and as incremental updates.

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py [hunk 1]
```diff
@@ -1031,27 +1036,6 @@ class Deals(CRMSearchStream):
     last_modified_field = "hs_lastmodifieddate"
     associations = ["contacts", "companies"]
 
-    def __init__(self, **kwargs):
-        super().__init__(**kwargs)
-        self._stage_history = DealStageHistoryStream(**kwargs)
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py:1036]: I think we need `DealStageHistoryStream` class from code, not only where it is used. It seems `DealStageHistoryStream` class was used only in `Deals` class.

**Review by lazebnyi — CHANGES_REQUESTED** [review]
LGTM. 
Small comment below.
Pls, add a changelog to `hubspot.md` file. And bump version in docker file.

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py (hunk 1 above)
Comment (lgomezm) [reply airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py:1036]: Removed in [8db98f2](https://github.com/airbytehq/airbyte/pull/10707/commits/8db98f2492064fa94868c42af2a59ae0ec977e8d).

**Comment by lgomezm** [comment]
> LGTM. Small comment below. Pls, add a changelog to `hubspot.md` file. And bump version in docker file.

@lazebnyi I've bumped the version in [5da84c8](https://github.com/airbytehq/airbyte/pull/10707/commits/5da84c80edb2d72e7d4711b62b6f781d6213a287).

**Review by lazebnyi — APPROVED** [review]

**Pull request merged**